src/gui/qgsrelationeditorwidget.cpp

```diff
@@ -198,10 +258,18 @@ void QgsRelationEditorWidget::setRelationFeature( const QgsRelation &relation, c
   if ( mVisible )
   {
     QgsFeatureRequest myRequest = mRelation.getRelatedFeaturesRequest( mFeature );
-    mDualView->init( mRelation.referencingLayer(), nullptr, myRequest, mEditorContext );
+    initDualView( mRelation.referencingLayer(), myRequest );
   }
 }
 
+void QgsRelationEditorWidget::initDualView( QgsVectorLayer *layer, const QgsFeatureRequest &request )
+{
+  mDualView->init( layer, nullptr, request, mEditorContext );
+  mFeatureSelectionMgr = new QgsFilteredSelectionManager( layer, request, mDualView );
```

Comment:
This iterates over the features twice, correct?
Once in the dual view and once in the feature selection mgr. I wonder if this could be reduced to one single time.

Reply:
Actually, there was and issue on the way I built the selected feature ids list b23a972
So now, it iterates only on selected feature list to check if the feature matches the relation.

I cannot a see a correct way to do this one single time.